### PR TITLE
WIP: Do whatever it takes to build osx

### PR DESCRIPTION
**Pull request by tmfset**
Build and run the emulator on OSX, it works!

I may have made some mistakes in the process though. Will have to review later.

### CODE REVIEW

File: scripts/linux.mk [hunk 1]
```diff
@@ -1,15 +1,15 @@
 # Build Tools for Linux
-CC := gcc -fdiagnostics-color -fmax-errors=5
-CPP := g++ -fdiagnostics-color -fmax-errors=5
+CC := gcc-10 -fdiagnostics-color -fmax-errors=5 -I/usr/local/Cellar/sdl2/2.0.14/include/ -I/usr/local/Cellar/sdl2_ttf/2.0.15/include/ -I/usr/local/Cellar/sdl2/2.0.14/include/SDL2/ -rdynamic
```
Comment (tmfset) [inline scripts/linux.mk:2]: We need a better way to source these includes.

File: scripts/emu.mk [hunk 2]
```diff
@@ -15,24 +15,24 @@ libraries += $(libs_build_dir)/liblodepng.a
...
+# Manually add objects
 objects += $(out_dir)/od/glue/app_swig.o
 objects += $(out_dir)/libs/SDL_FontCache/SDL_FontCache.o
- 
+
 CFLAGS += -DFIRMWARE_VERSION=\"$(FIRMWARE_VERSION)\"
 CFLAGS += -DFIRMWARE_NAME=\"$(FIRMWARE_NAME)\"
 CFLAGS += -DFIRMWARE_STATUS=\"$(FIRMWARE_STATUS)\"
-LFLAGS = -Wl,--export-dynamic -Wl,--gc-sections -lSDL2 -lSDL2_ttf -lfftw3f -lm -ldl -lstdc++ 
+LFLAGS = -L/usr/local/Cellar/sdl2/2.0.14/lib/ -L/usr/local/Cellar/sdl2_ttf/2.0.15/lib/ -L/usr/local/Cellar/fftw/3.3.9/lib/ -lSDL2 -lSDL2_ttf -lfftw3f -lm -ldl -lstdc++
```
Comment (tmfset) [inline scripts/emu.mk:30]: Same issue with the library paths as with the includes, gotta find a good way to source them.

File: scripts/linux.mk [hunk 3]
```diff
@@ -1,15 +1,15 @@
 # Build Tools for Linux
```
Comment (tmfset) [inline scripts/linux.mk:1]: Probably want to create a new file just for osx.

File: arch/linux/hal/fileops.c [hunk 4]
```diff
@@ -1,6 +1,7 @@
 #include <hal/fileops.h>
 #include <sys/stat.h>
-#include <sys/statfs.h>
+#include <sys/param.h>
```
Comment (tmfset) [inline arch/linux/hal/fileops.c:3]: Are these both actually required?

**Comment by odevices** [comment]
I was able to introduce some of the trivial changes.  Others (like the changes in the shared non-arch-specific code) will take a bit more work.

**Comment by tmfset** [comment]
@odevices I was thinking I'll add an `osx` arch target and associated files, what do you think?

**Comment by tmfset** [comment]
Closing, working on a clean PR to add OSX support to the makefile.

**Pull request closed without merging**